runtime/src/bank/partitioned_epoch_rewards/calculation.rs

```diff
@@ -336,98 +336,134 @@ impl Bank {
         } = reward_calculate_params;
 
         let new_warmup_cooldown_rate_epoch = self.new_warmup_cooldown_rate_epoch();
-        let estimated_num_vote_accounts = cached_vote_accounts.len();
-        let vote_account_rewards: VoteRewards = DashMap::with_capacity_and_hasher_and_shard_amount(
-            estimated_num_vote_accounts,
-            AHashRandomState::default(),
-            1024, // shard amount
-        );
 
-        let total_stake_rewards = AtomicU64::default();
+        let vote_account_rewards_len = Arc::new(AtomicUsize::default());
+        let total_stake_rewards = Arc::new(AtomicU64::default());
+        let reward_calc_tracer = Arc::new(reward_calc_tracer);
         const ASSERT_STAKE_CACHE: bool = false; // Turn this on to assert that all vote accounts are in the cache
-        let (stake_rewards, measure_stake_rewards_us) = measure_us!(thread_pool.install(|| {
-            stake_delegations
-                .par_iter()
-                .filter_map(|(stake_pubkey, stake_account)| {
-                    // curry closure to add the contextual stake_pubkey
-                    let reward_calc_tracer = reward_calc_tracer.as_ref().map(|outer| {
-                        // inner
-                        move |inner_event: &_| {
-                            outer(&RewardCalculationEvent::Staking(stake_pubkey, inner_event))
-                        }
-                    });
 
-                    let stake_pubkey = **stake_pubkey;
-                    let vote_pubkey = stake_account.delegation().voter_pubkey;
-                    let vote_account_from_cache = cached_vote_accounts.get(&vote_pubkey);
-                    if ASSERT_STAKE_CACHE && vote_account_from_cache.is_none() {
-                        let account_from_db = self.get_account_with_fixed_root(&vote_pubkey);
-                        if let Some(account_from_db) = account_from_db {
-                            if VoteStateVersions::is_correct_size_and_initialized(
-                                account_from_db.data(),
-                            ) && VoteAccount::try_from(account_from_db.clone()).is_ok()
-                            {
-                                panic!(
-                                    "Vote account {} not found in cache, but found in db: {:?}",
-                                    vote_pubkey, account_from_db
+        let num_workers = num_cpus::get();
+
+        // Collections gathered in the pool.
+        let mut stake_rewards: Vec<_> = (0..num_workers)
+            .map(|_| Vec::with_capacity(stake_delegations.len().div_ceil(num_workers)))
+            .collect();
+        let mut vote_account_rewards: Vec<_> = (0..num_workers)
```

Comment:
What if two hashmaps contain the vtoe rewards for the same vote account? I think we will need to merge those hashmaps after computing.

Reply:
sounds good.

once you fix this, I can review it again.